### PR TITLE
feat(mysql,postgresql): make maintenance window optional

### PR DESCRIPTION
**Pull request by chicocvenancio**
This fixes issue #64.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@chicocvenancio
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/mysql/main.tf
--- old/modules/mysql/main.tf
+++ new/modules/mysql/main.tf
@@ -123,7 +123,7 @@
       follow_gae_application = var.follow_gae_application
     }
 
-    dynamic maintenance_window {
+    dynamic "maintenance_window" {
       for_each = var.maintenance_window_enabled ? ["enabled"] : []
       content {
         day          = var.maintenance_window_day
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/postgresql/main.tf
--- old/modules/postgresql/main.tf
+++ new/modules/postgresql/main.tf
@@ -127,7 +127,7 @@
       follow_gae_application = var.follow_gae_application
     }
 
-    dynamic maintenance_window {
+    dynamic "maintenance_window" {
       for_each = var.maintenance_window_enabled ? ["enabled"] : []
       content {
         day          = var.maintenance_window_day
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@chicocvenancio
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/mysql/main.tf
--- old/modules/mysql/main.tf
+++ new/modules/mysql/main.tf
@@ -123,7 +123,7 @@
       follow_gae_application = var.follow_gae_application
     }
 
-    dynamic maintenance_window {
+    dynamic "maintenance_window" {
       for_each = var.maintenance_window_enabled ? ["enabled"] : []
       content {
         day          = var.maintenance_window_day
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/postgresql/main.tf
--- old/modules/postgresql/main.tf
+++ new/modules/postgresql/main.tf
@@ -127,7 +127,7 @@
       follow_gae_application = var.follow_gae_application
     }
 
-    dynamic maintenance_window {
+    dynamic "maintenance_window" {
       for_each = var.maintenance_window_enabled ? ["enabled"] : []
       content {
         day          = var.maintenance_window_day
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@chicocvenancio
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/mysql/variables.tf [hunk 1]
```diff
@@ -115,6 +115,12 @@ variable "pricing_plan" {
   default     = "PER_USE"
 }
 
+variable "maintenance_window_enabled" {
+  description = "Whether to enable maintenance window setting for the instance. Replicas must have this set to false."
```
Comment (g-awmalik) [inline modules/mysql/variables.tf:119]: Can you rename this slightly as:
"Whether to enable maintenance window setting for the instance. This does not apply to replicas."

**Review by g-awmalik — APPROVED** [review]
Thanks for the PR. One minor comment.

**Comment by kpocius** [comment]
@g-awmalik, can you sort out Cloud Build runs? It says "Needs /gcbrun from a collaborator" ¯\_(ツ)_/¯

**Comment by g-awmalik** [comment]
@kpocius - can you rebase this on master and push again? Seems like it's pulling in historical commits that don't need to be there.

**Comment by kpocius** [comment]
@g-awmalik, all good now? 🤔

**Comment by g-awmalik** [comment]
@kpocius - seems like your branch is protected and I'm unable to rebase it with master. Can you do that or unprotect your branch so I can do that for you?

**Comment by kpocius** [comment]
@g-awmalik, that's odd -- there are definitely no branch protection rules, but maybe some other setting is preventing you from doing a rebase. Either way, I've just rebased it.

**Comment by g-awmalik** [comment]
@apeabody - do you have access to rebase this with master?

**Comment by apeabody** [comment]
> @apeabody - do you have access to rebase this with master?

I do not, this might not be set on the fork: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by kpocius** [comment]
I'm not the original author of this PR, so I can't seem to be able to set "Allow edits from maintainers", but I can rebase it.

**Comment by kpocius** [comment]
Seems like that option is not available when the repo is forked to an organization https://github.com/orgs/community/discussions/5634

**Comment by apeabody** [comment]
> Seems like that option is not available when the repo is forked to an organization https://github.com/orgs/community/discussions/5634

Good to know!

**Comment by g-awmalik** [comment]
@kpocius - we'll need this rebased again since it's out of date.

**Comment by g-awmalik** [comment]
Also, documentation seems to be out of date. Please run `make generate_docs` with the rebased code once again.

**Pull request closed without merging**